lib/std/start.zig

```diff
@@ -224,14 +224,27 @@ fn EfiMain(handle: uefi.Handle, system_table: *uefi.tables.SystemTable) callconv
 }
 
 fn _start() callconv(.Naked) noreturn {
+    // It's important that all checks here are comptime since having any kind of non-asm code in a
+    // naked function is asking for trouble with some LLVM backends.
+
     // TODO set Top of Stack on non x86_64-plan9
-    if (native_os == .plan9 and native_arch == .x86_64) {
+    if (comptime native_os == .plan9 and native_arch == .x86_64) {
```

Comment:
the comptime is obviously redundant. there's nothing special making this conditional compilation more important than the other thousands of instances of conditional compilation.

```suggestion
    if (native_os == .plan9 and native_arch == .x86_64) {
```

Reply:
I did it for 2 reasons:

1. Consistency with the check below for riscv which does need `comptime`.
2. So it doesn't unexpectedly turn into a bug if someone edits the condition to include a call to an `inline` function.

That said, if #21070 is acceptable, I don't think the `comptime` is necessary on the riscv check either.